src/coreclr/jit/importercalls.cpp

```diff
@@ -662,8 +670,57 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
         }
 
         checkForSmallType = true;
+        bool spillAllArgs = false;
+
+#ifdef SWIFT_SUPPORT
+        // We are importing an unmanaged Swift call, which might require special parameter handling:
+        // - SwiftError* is passed to capture the address of an error thrown during the Swift call.
+        if (call->AsCall()->unmgdCallConv == CorInfoCallConvExtension::Swift)
+        {
+            // Check the signature of the Swift call for the special types.
+            CORINFO_ARG_LIST_HANDLE sigArg = sig->args;
+
+            for (unsigned short argIndex = 0; argIndex < sig->numArgs;
+                 sigArg                  = info.compCompHnd->getArgNext(sigArg), argIndex++)
+            {
+                CORINFO_CLASS_HANDLE argClass;
+                CorInfoType          argType         = strip(info.compCompHnd->getArgType(sig, sigArg, &argClass));
+                bool                 argIsByrefOrPtr = false;
+
+                if ((argType == CORINFO_TYPE_BYREF) || (argType == CORINFO_TYPE_PTR))
+                {
+                    argClass        = info.compCompHnd->getArgClass(sig, sigArg);
+                    argType         = info.compCompHnd->getChildType(argClass, &argClass);
+                    argIsByrefOrPtr = true;
+                }
 
-        impPopArgsForUnmanagedCall(call->AsCall(), sig);
+                if ((argType != CORINFO_TYPE_VALUECLASS) || !info.compCompHnd->isIntrinsicType(argClass))
+                {
+                    continue;
+                }
+
+                const char* namespaceName;
+                const char* className = info.compCompHnd->getClassNameFromMetadata(argClass, &namespaceName);
+
+                if ((strcmp(className, "SwiftError") == 0) &&
+                    (strcmp(namespaceName, "System.Runtime.InteropServices.Swift") == 0))
+                {
+                    // For error handling purposes, we expect a pointer to a SwiftError to be passed
+                    assert(argIsByrefOrPtr);
+                    if (swiftErrorIndex != -1)
+                    {
+                        BADCODE("Duplicate SwiftError* parameter");
+                    }
+
+                    swiftErrorIndex = argIndex;
+                    spillAllArgs    = true;
+                }
+                // TODO: Handle SwiftSelf, SwiftAsync
+            }
+        }
+#endif // SWIFT_SUPPORT
+
+        impPopArgsForUnmanagedCall(call->AsCall(), sig, spillAllArgs);
```

Comment:
I think it would be better to move all of this handling into `impPopArgsForUnmanagedCall`.

Reply:
I wanted to do that initially, but I'm not sure what the best way is to communicate which CallArgs correspond to special Swift types back to `impImportCall`. Are you ok with `impPopArgsForUnmanagedCall` having some out params that are only used in the Swift case?

Reply:
Yeah, I think that sounds just fine.